### PR TITLE
Enable plotting of individual functions in FitPropertyBrowser 

### LINKED ISSUE #29622: Plotting individual functions from fit property browser
### Expected behavior
Expect that the capability to plot an individual function when fitting.

### Actual behavior
Plot option does not work on right-click menu of function of fit property browser

### Steps to reproduce the behavior
(1) Load workspace
(2) Plot
(3) Toggle fit
(4) add a function
(5) right-click on function in fitproperty browser (see screenshot)
(6) plot

### Platforms affected
at least since v5.0

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR allows the user to plot individual functions in the fit property browser. This replicates a feature which was present in MantidPlot. 

**To test:**
- Open workbench
- Load some data
- Plot and open the fit browser
- Add any three functions
- Plot each one by right clicking on the function
- Delete the first function
- The plot should update

Generally, just play around with the fit browser and make sure everything works.

Fixes #29622 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v6.0.0/mantidworkbench.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ Improvements
 ############
 
 - Enable workbench to plot 1D spectrum from an IMDHistoWorkspace whose non-integrated dimension is equal to 1 but not to launch slice viewer.
+- Enabled the plotting of individual functions in the Workbench fit browser. This replicates a feature that was in MantidPlot
```
Comment (robertapplin) [inline docs/source/release/v6.0.0/mantidworkbench.rst:12]: Would it be good to mention how to us this new feature (i.e. by Right-clicking a function in the browser and selecting `Plot`)?

File: qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        Evaluates the guess workspace for the input workspace and function
+        :param ws_name: Name of the workspace in the ADS
+        :param fun: Function to be evaluated
+        :param out_ws_name: Output workspace name
+        :return: Output guess workspace
+        """
+        ws_index = self.fit_browser.workspaceIndex()
+        startX = self.fit_browser.startX()
+        endX = self.fit_browser.endX()
+        workspace = AnalysisDataService.retrieve(ws_name)
```
Comment (robertapplin) [inline qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py:125]: Would a check to see if the workspace exists in the ADS before this line be important here?

File: qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+    def plot_guess_all(self):
+        """
+        Plot the guess workspace for the full function
+        """
+        fun = self.fit_browser.getFittingFunction()
+        ws_name = self.fit_browser.workspaceName()
+        if fun == '' or ws_name == '':
+            return
+        out_ws_name = '{}_guess'.format(ws_name)
```
Comment (robertapplin) [inline qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py:153]: Could we use [f-strings](https://realpython.com/python-f-strings/) throughout here for better readability?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Plotting individual functions and removing them works as expected. Code changes look good, just a few suggestions.

File: qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py (hunk 2 above)
Comment (StephenSmith25) [reply qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py:125]: As this is the same workspace thats in the plot we can be pretty certain it exists. If its not in the ads (e.g if we delete it while having a plot shown) it appears to close the plot.

File: qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py (hunk 3 above)
Comment (StephenSmith25) [reply qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py:153]: Yep, good point 👍

**Review by robertapplin — APPROVED** [review]

**Pull request merged**